src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -382,180 +379,171 @@ void GetZoneEquipmentData(EnergyPlusData &state)
-                    ShowContinueError(state,
-                                      format("invalid {}=[{}].",
-                                             cNumericFields(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 1),
-                                             thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum)));
-                    ShowContinueError(state, "equipment sequence must be > 0 and <= number of equipments in the list.");
-                    if (thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum) > 0)
-                        ShowContinueError(state, format("only {} in the list.", thisZoneEquipList.NumOfEquipTypes));
-                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
-                }
+            UtilityRoutines::IsNameEmpty(state, thisZoneEquipList.Name, CurrentModuleObject, state.dataZoneEquip->GetZoneEquipmentDataErrorsFound);
 
-                thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum) =
-                    nint(NumArray(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 2));
-                if ((thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum) < 0) ||
-                    (thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum) > thisZoneEquipList.NumOfEquipTypes)) {
-                    ShowSevereError(state, std::string{RoutineName} + CurrentModuleObject + "=\"" + AlphArray(1) + "\".");
-                    ShowContinueError(state,
-                                      format("invalid {}=[{}].",
-                                             cNumericFields(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 2),
-                                             thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum)));
-                    ShowContinueError(state, "equipment sequence must be > 0 and <= number of equipments in the list.");
-                    if (thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum) > 0)
-                        ShowContinueError(state, format("only {} in the list.", thisZoneEquipList.NumOfEquipTypes));
-                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
-                }
+            std::string loadDistName = ip->getAlphaFieldValue(epListFields, objectSchemaProps, "load_distribution_scheme");
+            thisZoneEquipList.LoadDistScheme = static_cast<DataZoneEquipment::LoadDist>(
+                getEnumerationValue(DataZoneEquipment::LoadDistNamesUC, UtilityRoutines::MakeUPPERCase(loadDistName)));
+            if (thisZoneEquipList.LoadDistScheme == DataZoneEquipment::LoadDist::Invalid) {
+                ShowSevereError(state, format("{}{} = \"{}, Invalid choice\".", RoutineName, CurrentModuleObject, thisZoneEquipList.Name));
+                ShowContinueError(state, format("...load_distribution_scheme=\"{}\".", loadDistName));
+                state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
+            }
```

Comment:
regular field, with getEnumeration value.

Reply:
This check is not needed. It is not possible for this field to be Invalid. If this field is blank the Json input processor will fill the field with SequentialLoad. If it's not a valid key choice the Json processor will fatal out.

    ZoneHVAC:EquipmentList,
      A2 , \field Load Distribution Scheme
           \type choice
           \key SequentialLoad
           \key UniformLoad
           \key UniformPLR
           \key SequentialUniformPLR
           \default SequentialLoad

Reply:
Oh, but this is an extensible field so I am not sure what the Json input processor will do with the extended fields. I would think it would process these extended fields the same way as those that are within min-fields.

Reply:
> If this field is blank the Json input processor will fill the field with SequentialLoad.

I don't think that's so. `getObjectItem` fills in defaults (up to min-fields), and `getAlphaFieldValue` etc. fills in defaults. Raw epJSON will simply not have a value for this field.